wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -219,12 +227,13 @@ public void configurePipeline(PipelineConfigurer configurer) {
       }
 
       // Check if pre-condition is not null or empty and if so compile expression.
-      if (!config.containsMacro(Config.NAME_PRECONDITION)) {
-        if (config.precondition != null && !config.precondition.trim().isEmpty()) {
+      if (!config.containsMacro(Config.NAME_PRECONDITION_JEXL)) {
+        if (config.preconditionLanguage == "jexl" && config.preconditionJexl != null
```

Comment:
Since this is a string it should use `.equals()` and not `==`. Also, can you please move this long boolean condition into a function so that it is more readable here? And also can you add a similar function for checking if the SQL input is valid?